tests/unit/#72.test.scala

```diff
@@ -0,0 +1,26 @@
+// SYNTAX TEST "source.scala"
+
+   s"$first$second"
+// ^ keyword.interpolation.scala
+//  ^ punctuation.definition.string.begin.scala
+//   ^ punctuation.definition.template-expression.begin.scala
+//         ^ punctuation.definition.template-expression.begin.scala
+//                ^ punctuation.definition.string.end.scala
+
+   s"$safeTagMarker${mtch.matched}$safeTagMarker"
+// ^ keyword.interpolation.scala
+//  ^ punctuation.definition.string.begin.scala
+//   ^ punctuation.definition.template-expression.begin.scala
+//                 ^^ punctuation.definition.template-expression.begin.scala
+//                   ^^^^^^^^^^^^ punctuation.definition.template-expression.scala
```

Comment:
This part should be parsed as an arbitrary Scala block.  For example, it could contain definitions

```scala
val a = 4; foo(a)
s"$safeTagMarker${val a = 4; foo(a)}$safeTagMarker"
```

Both `val a = 4; foo(a)` should have the same highlighting.

Reply:
Oh right I can see the `foo`s have different highlighting here.
<img width="377" alt="Screen Shot 2020-01-27 at 1 04 10 PM" src="https://user-images.githubusercontent.com/6521018/73151296-934a8d80-4105-11ea-818c-fbf82f545f15.png">

The issue is `val a = 4; foo(a)` has scope `punctuation.definition.template-expression.scala` inside string interpolation and since `; foo(a)` has no scope it was highlighted the same way as `${` and `}` (`punctuation.definition.template-expression.begin.scala` and `punctuation.definition.template-expression.end.scala`).

I just pushed another commit to fix this. Basically replace `punctuation.definition.template-expression.scala` with `meta.embedded.line.scala` (Inspiration from typescript and ruby tm highlighting). More details in the commit message which I'm just gonna paste here.

> Add scopes `meta.template.expression.scala` and `meta.embedded.line.scala`
Eg. In `s"${val a = 4; foo(a)}"`, `${val a = 4; foo(a)}` will have `meta.template.expression.scala` while `val a = 4; foo(a)` will have `meta.embedded.line.scala`.
Earlier `val a = 4; foo(a)` was under `punctuation.definition.template-expression`. This makes things without a scope like `; foo(a)` highlighted differently inside string interpolation (they will have the same highlighting as `${` and `}`) vs outside, which is not desirable.


Visual tests

Before
<img width="377" alt="Screen Shot 2020-01-27 at 1 04 10 PM" src="https://user-images.githubusercontent.com/6521018/73151596-c9d4d800-4106-11ea-94c4-3d68beadfb54.png">
After
<img width="379" alt="Screen Shot 2020-01-27 at 12 40 36 PM" src="https://user-images.githubusercontent.com/6521018/73151605-d6593080-4106-11ea-94da-b7b0707ad340.png">

Before
<img width="342" alt="Screen Shot 2020-01-27 at 11 57 27 AM" src="https://user-images.githubusercontent.com/6521018/73151635-fd176700-4106-11ea-8d9e-c96b4e177f2f.png">
After
<img width="340" alt="Screen Shot 2020-01-27 at 12 40 42 PM" src="https://user-images.githubusercontent.com/6521018/73151646-07d1fc00-4107-11ea-9b6b-983c6a48bdbb.png">